### PR TITLE
Feature/fix roots

### PR DESCRIPTION
**Pull request by RickyVimon**
What's new?

- New logic for the roots. Now the roots vector is aligned with the Zones vector; So, roots_obstacles[3] will return the root Game Object on the Zone 3. Will return nullptr if there are no roots on that area. Nullptr checks have been set to control any crash.
- InitRoots is done together with the InitZones.
- Ingame now if the player respwans on a checkpoint, all the roots which have been passed will be down.
- Earthquake sound fixed with respawning.

### CODE REVIEW

File: Scripts/src/Script/EventManager.h [hunk 1]
```diff
@@ -120,7 +120,7 @@ class EventManager : public Script
 
 	//Root events
 	std::vector<GameObject*> root_obstacles;
-	int current_root = 0;
+	//std::vector<GameObject*> root_zones;
```
Comment (Dartemiss) [inline Scripts/src/Script/EventManager.h:123]: why is comented?

**Review by Dartemiss — CHANGES_REQUESTED** [review]
Work fine except the sound, If you finish zone 3 of level1 and purify everything the sound becomes purified when respawning.

File: Scripts/src/Script/EventManager.h (hunk 1 above)
Comment (RickyVimon) [reply Scripts/src/Script/EventManager.h]: solved

**Review by Pacasasgar — CHANGES_REQUESTED** [review]
I tested your branch:

1. Roots don't reproduce sound when you purify them, at least after beating battle zone 1, so first roots (I can't hear the 'earthquake' sound).

2. If I die in the River and re-spawn after battle zone 2, roots are there again behind me.

PD: Which scene should I be using to test? Level1.scene or Level1_roots.scene? I tried both of them and above issues still happened so maybe I'm just missing something.

**Comment by RickyVimon** [comment]
> I tested your branch:
> 
> 1. Roots don't reproduce sound when you purify them, at least after beating battle zone 1, so first roots (I can't hear the 'earthquake' sound).
> 2. If I die in the River and re-spawn after battle zone 2, roots are there again behind me.
> 
> PD: Which scene should I be using to test? Level1.scene or Level1_roots.scene? I tried both of them and above issues still happened so maybe I'm just missing something.

I have just tested it and it works on my branch. Are you sure you are up to date to develop in the engine and on the feature/fix_roots branch?

**Comment by Pacasasgar** [comment]
> > I tested your branch:
> > 
> > 1. Roots don't reproduce sound when you purify them, at least after beating battle zone 1, so first roots (I can't hear the 'earthquake' sound).
> > 2. If I die in the River and re-spawn after battle zone 2, roots are there again behind me.
> > 
> > PD: Which scene should I be using to test? Level1.scene or Level1_roots.scene? I tried both of them and above issues still happened so maybe I'm just missing something.
> 
> I have just tested it and it works on my branch. Are you sure you are up to date to develop in the engine and on the feature/fix_roots branch?

As discussed on discord, I will approve now the PR. Regarding my problems:

1. It is not an issue caused by thir PR changes. It's a new bug probably caused by the implementation of the new XML parser in the soundbank events. This will be fixed in another branch.

2. It was working fine if you play normally the game. The problem I described appears when using 'F1' cheat mode to teleport to different zones. You appear behind the roots.

**Review by Pacasasgar — APPROVED** [review]

**Review by Dartemiss — APPROVED** [review]
LGTM

**Pull request merged**